internal/acceptance/notification_destination_test.go

```diff
@@ -0,0 +1,217 @@
+package acceptance
+
+import (
+	"context"
+	"testing"
+
+	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/databricks/terraform-provider-databricks/qa"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func check(t *testing.T, ctx context.Context, client *common.DatabricksClient, id, display_name string, config_type settings.DestinationType) error {
+	w, err := client.WorkspaceClient()
+	if err != nil {
+		return err
+	}
+	ndResource, err := w.NotificationDestinations.Get(ctx, settings.GetNotificationDestinationRequest{
+		Id: id,
+	})
+	if err != nil {
+		return err
+	}
+	assert.Equal(t, config_type, ndResource.DestinationType)
+	assert.Equal(t, display_name, ndResource.DisplayName)
+	require.NoError(t, err)
+	return nil
+}
```

Comment:
It's not critical, but yon can further simplify the code by returning a function, so instead of

```
		Check: resourceCheck("databricks_notification_destination.this", func(ctx context.Context, client *common.DatabricksClient, id string) error {
			return check(t, ctx, client, id, display_name, settings.DestinationTypeEmail)
		}),
```

you can write something like:

```
Check: makeCheckFunction(dsetinationType, displayName)
```